### PR TITLE
Add raft settings to make margin and remove inside corners configurablle per raft-type

### PR DESCRIPTION
**Pull request by casperlamboo**
# Description

Added the following settings
- raft_base_margin
- raft_interface_margin
- raft_surface_margin
- raft_base_smoothing
- raft_interface_smoothing
- raft_surface_smoothing
- raft_base_remove_inside_corners
- raft_interface_remove_inside_corners
- raft_surface_remove_inside_corners

Screenshot 2024-01-08 at 14 48 21

CURA-11395
Also see backend PR: https://github.com/Ultimaker/CuraEngine/pull/2006

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Printer definition file(s)
- [ ] Translations

# How Has This Been Tested?

**Test Configuration**:
* Operating System:
macOS 13.3

# Checklist:

- [x] My code follows the style guidelines of this project as described in [UltiMaker Meta](https://github.com/Ultimaker/Meta) and [Cura QML best practices](https://github.com/Ultimaker/Cura/wiki/QML-Best-Practices)
- [x] I have read the [Contribution guide](https://github.com/Ultimaker/Cura/blob/main/CONTRIBUTING.md) 
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have uploaded any files required to test this change

### CODE REVIEW

**Review by wawanbreton — CHANGES_REQUESTED** [review]

File: resources/definitions/fdmprinter.def.json [hunk 1]
```diff
@@ -6114,7 +6114,55 @@
...
+                            "label": "Raft Middle Extra Margin",
+                            "description": "If the raft middle is enabled, this is the extra raft area around the model which is also given a raft. Increasing this margin will create a stronger raft while using more material and leaving less area for your print.",
+                            "unit": "mm",
+                            "type": "float",
+                            "value": "raft_margin",
+                            "default_value": 15,
+                            "minimum_value_warning": "raft_interface_line_width",
+                            "maximum_value_warning": "20",
+                            "enabled": "resolveOrValue('adhesion_type') == 'raft'",
+                            "limit_to_extruder": "adhesion_extruder_nr",
```
[the remark this answers is not in the document]
Comment (casperlamboo) [reply resources/definitions/fdmprinter.def.json]: https://github.com/Ultimaker/Cura/pull/17945/commits/37f350b134d69518ef6455f214f152e4e04737b6

File: resources/definitions/fdmprinter.def.json [hunk 2]
```diff
@@ -8332,7 +8428,46 @@
                     "resolve": "any(extruderValues('raft_remove_inside_corners'))",
                     "enabled": "resolveOrValue('adhesion_type') == 'raft'",
                     "settable_per_mesh": false,
-                    "settable_per_extruder": false
+                    "settable_per_extruder": false,
+                    "children":
```
[the remark this answers is not in the document]
Comment (casperlamboo) [reply resources/definitions/fdmprinter.def.json]: https://github.com/Ultimaker/Cura/pull/17945/commits/af51ccf5f37c5fdaf93bf31de307c5d3678f3691

File: resources/setting_visibility/expert.cfg [hunk 3]
```diff
@@ -468,3 +474,6 @@ small_hole_max_size
 small_feature_max_length
 small_feature_speed_factor
 small_feature_speed_factor_0
+raft_base_remove_inside_corners
+raft_interface_remove_inside_corners
+raft_surface_remove_inside_corners
```
[the remark this answers is not in the document]
Comment (casperlamboo) [reply resources/setting_visibility/expert.cfg]: https://github.com/Ultimaker/Cura/pull/17945/commits/4489b8db462e28777e24947fbcc09517566ecfe2

File: resources/definitions/fdmprinter.def.json [hunk 4]
```diff
@@ -6114,7 +6114,55 @@
...
+                        "raft_surface_margin":
+                        {
+                            "label": "Raft Top Extra Margin",
+                            "description": "If the raft top is enabled, this is the extra raft area around the model which is also given a raft. Increasing this margin will create a stronger raft while using more material and leaving less area for your print.",
+                            "unit": "mm",
+                            "type": "float",
+                            "value": "raft_margin",
+                            "default_value": 15,
+                            "minimum_value_warning": "raft_interface_line_width",
+                            "maximum_value_warning": "20",
```
[the remark this answers is not in the document]
Comment (casperlamboo) [reply resources/definitions/fdmprinter.def.json:6159]: https://github.com/Ultimaker/Cura/pull/17945/commits/594e4a5ea67fc0266783dfe8052f2b3a8b0191cc

**Pull request merged**